### ISSUE TITLE
Add colored headers and odd/even rows

### ISSUE THREAD

**Issue by mikeerickson** [dbcli/mycli#364]
In short, it would be swell if the output could provide some colors.   I dont think it needs to be elaborate, thinking the colors could be added the `~/.myclirc` 

Ala...

```
# Colors
Token.Table.Header = 'bg:#222222 #888888'
Token.Table.OddRow = 'bg:#222222 #ffffff'
Token.Table.EvenRow = 'bg:#222222 #ffffff'
```

**Comment by tsroten** [comment]
@mikeerickson Thanks for using mycli!

I'm curious about how you envision this working with the ASCII table formats:
```
+--------+--------+
| col1   | col2   |
|--------+--------|
| row1   | row1   |
|--------+--------|
| row2   | row2   |
+--------+--------+
```

For example, if a row is supposed be red, would that affect the values only, or also the `|` characters that separate the columns?

**Comment by mikeerickson** [comment]
I have created numerous CLI tools (for node development in my case) and I often output data as CLI table (see screenshot)

![image](https://cloud.githubusercontent.com/assets/183153/23769388/1401de32-04c4-11e7-99b7-fab83300cc8a.png)

**Comment by mikeerickson** [comment]
@tsroten In this case, if just the header were colorized, that would be sufficient.  But figured, while there, how about supporting add/even row.

In all cases, simply coloring the foreground would be sufficient

**Comment by tsroten** [comment]
Great, thanks for the input!

Here are some initial thoughts for anyone interested in giving this a shot.
- We currently output text using `click.secho()` for the entire set of formatted results at once. That function does support colored output, but it would be applied to the entire outputted text.
- Instead, we might be able to feed each value through `click.style()` (which adds color codes to the text) prior to putting it in a table format (maybe in `mycli.main.format_output()`).
- [`click` supports](http://click.pocoo.org/5/api/#click.style) the typical 8 ANSI colors for the foreground.

**Comment by mikeerickson** [comment]
ping @tsroten :-)

**Comment by tsroten** [comment]
Hi @mikeerickson! I'm not sure when I'll be able to get to this. Pull requests are welcome from anyone :)

**Comment by tsroten** [comment]
Ok, initial progress on this is looking good. The challenge now is figuring out how to handle writing the output to files. A given query's output has the potential to be written to an audit log file, tee file, and one-time output/export file.

In the files we don't want the escape codes, but we don't want to have to always strip the escape codes.

<img width="227" alt="screen shot 2017-06-09 at 09 03 31" src="https://user-images.githubusercontent.com/3047055/26978874-8ffe8b86-4cf2-11e7-87cd-c1ab0d256fce.png">

**Comment by amjith** [comment]
I wonder if click provides a way to pipe an output where it strips the color codes.

**Comment by j-bennet** [comment]
@tsroten @amjith It looks like you can provide a `file` parameter to `click.echo` and it will strip ANSI color codes for you:

https://github.com/pallets/click/blob/945b4d38a323e8957df04ca371e7e032f930179e/click/utils.py#L167-L261

It does the stripping if the provided `file` is not a TTY:

https://github.com/pallets/click/blob/a50befdc3ac475826af75e48fd48b21530851176/click/_compat.py#L520-L529

* Note to self - I should probably borrow that stripping call instead of using my own regex in `behave` tests.

**Comment by tsroten** [comment]
@j-bennet Good job noticing that! I've added the `click.echo` calls to the pull request (depends on new CLI Helpers release, so the tests fail right now).

**Comment by tsroten** [comment]
@mikeerickson This is included as of [version 1.11.0](https://pypi.org/project/mycli/). If you installed via `pip`, you can upgrade by running `pip install --upgrade mycli`.

Thanks for the feature request!
